bleach/sanitizer.py

```diff
@@ -16,6 +16,9 @@ class BleachSanitizerMixin(HTMLSanitizerMixin):
 
     allowed_svg_properties = []
 
+    def _has_unmatched_tag(self):
+        return self.strip_disallowed_element_contents and len(self.unmatched_tags) > 0
+
     def sanitize_token(self, token):
```

Comment:
I think _L40_ should be `if token['name'] in self.allowed_elements and not self._has_unmatched_tag():`. What do you think?